seisbench/models/base.py

```diff
@@ -1829,6 +1830,9 @@ def _predict_buffer(self, buffer):
         finally:
             if train_mode:
                 self.train()
+        if torch.cuda.is_available():
+            torch.cuda.synchronize()
```

Comment:
idk if this is in-line with the rest of the cuda handling.

The `synchronize` statement waits for all kernels to finish and helped profiling the inference call stack.

Reply:
I'm not sure if this synchronization is actually desired here. In general, torch will anyhow synchronize everything it needs right now. Does it have a performance advantage? If not, I'd suggest commenting it out and adding a note that this might be useful to add in when profiling the code.

Reply:
done